apps/studio/src/pages/sites/[siteId]/pages/[pageId]/index.tsx

```diff
@@ -169,7 +178,7 @@ const PageSettings = ({
   })
 
   const onSubmit = handleSubmit((values) => {
-    if (formState.isDirty) {
```

Comment:
i think this one is one render cycle behind because formState is a proxy. Might be able to get the latest state if this is invoked before the render instead of inside this function:

```ts
const {
  formState: { isDirty } // invoked here
  // ...
} = useZodForm(...)
```